src/SpacetimeDBNetworkManager.cs

```diff
@@ -6,38 +6,48 @@
 
 namespace SpacetimeDB
 {
-	// This class is only used in Unity projects.
-	// Attach this to a gameobject in your scene to use SpacetimeDB.
-	public class SpacetimeDBNetworkManager : MonoBehaviour
-	{
-		private static bool _alreadyInitialized;
+    // This class is only used in Unity projects.
+    // Attach this to a GameObject in your scene to use SpacetimeDB.
+    public class SpacetimeDBNetworkManager : MonoBehaviour
+    {
+        private static SpacetimeDBNetworkManager? _instance;
 
-		public void Awake()
-		{
-			// Ensure that users don't create several SpacetimeDBNetworkManager instances.
-			// We're using a global (static) list of active connections and we don't want several instances to walk over it several times.
-			if (_alreadyInitialized)
-			{
-				throw new InvalidOperationException("SpacetimeDBNetworkManager is a singleton and should only be attached once.");
-			}
-			else
-			{
-				_alreadyInitialized = true;
-			}
-		}
+        public void Awake()
+        {
+            // Ensure that users don't create several SpacetimeDBNetworkManager instances.
+            // We're using a global (static) list of active connections and we don't want several instances to walk over it several times.
+            if (_instance != null)
+            {
+                throw new InvalidOperationException("SpacetimeDBNetworkManager is a singleton and should only be attached once.");
+            }
+            else
+            {
+                _instance = this;
+            }
+        }
 
-		internal static HashSet<IDbConnection> ActiveConnections = new();
+        internal static HashSet<IDbConnection> ActiveConnections = new();
 
-		private void ForEachConnection(Action<IDbConnection> action)
-		{
-			foreach (var conn in ActiveConnections)
-			{
-				action(conn);
-			}
-		}
+        private List<IDbConnection> cache = new List<IDbConnection>();
 
-		private void Update() => ForEachConnection(conn => conn.FrameTick());
-		private void OnDestroy() => ForEachConnection(conn => conn.Disconnect());
-	}
+        private void ForEachConnection(Action<IDbConnection> action)
+        {
+            // TODO(jdetter): We're doing this for now because we can't break the API during a minor release but
+            // in the future we should just change ActiveConnections to be a list so that we can reverse-iterate
```

Comment:
ActiveConnections is already not part of the public API though? Or do you mean some other API?

Reply:
I *thought* that ActiveConnections was referenced in the bindings somewhere and I was worried about creating an API break there, but I grep'd in the SpacetimeDB project for `ActiveConnections` and there's nothing there so maybe you're right. Thanks for the heads up 👍

Reply:
It has `internal` visibility so bindings couldn't use it even if they wanted to.